### PR TITLE
Add `Tensor.T` attribute to reverse dimensions

### PR DESCRIPTION
**Pull request by umanwizard**
For compatibility with numpy

### CODE REVIEW

**Comment by umanwizard** [comment]
cc @SsnL

**Comment by umanwizard** [comment]
Ping -- can someone review this?

File: torch/csrc/autograd/python_variable.cpp [hunk 1]
```diff
@@ -159,6 +159,19 @@ static PyObject* THPVariable_make_subclass(PyObject* _ignored, PyObject* args, P
...
+PyObject *THPVariable_get_T(THPVariable *self)
+{
+  HANDLE_TH_ERRORS
+  auto& var = self->cdata;
+  int64_t n = var.dim();
+  DimVector transpose_dims;
+  for (int64_t i = n - 1; i >= 0; --i) {
+    transpose_dims.push_back(i);
+  }
+  return THPVariable_Wrap(var.permute(transpose_dims));
```
Comment (fmassa) [inline torch/csrc/autograd/python_variable.cpp]: Does this work for scalars, like `torch.tensor(5)`?

Comment (umanwizard) [reply torch/csrc/autograd/python_variable.cpp]: @fmassa Yes it works fine. It is a no-op, same as Numpy.

```python
>>> t=torch.tensor(5)
>>> t.T
tensor(5)
>>> a=np.array(5)
>>> a.T
array(5)
```

Comment (umanwizard) [reply torch/csrc/autograd/python_variable.cpp]: I will add a test for this case.

**Comment by umanwizard** [comment]
Wtf. I screwed something up -- didn't mean to close this.

**Review by fmassa — APPROVED** [review]

File: torch/csrc/autograd/python_variable.cpp [hunk 2]
```diff
@@ -159,6 +159,19 @@ static PyObject* THPVariable_make_subclass(PyObject* _ignored, PyObject* args, P
 typedef PyObject *(*getter)(PyObject *, void *);
 typedef int (*setter)(PyObject *, PyObject *, void *);
 
+PyObject *THPVariable_get_T(THPVariable *self)
```
Comment (gchanan) [inline torch/csrc/autograd/python_variable.cpp]: We don't have the ability to do this in C++, yet?  Can we add support for it?

Comment (umanwizard) [reply torch/csrc/autograd/python_variable.cpp]: Sure, what would you like to call it? `reverse_dims` ? Or just T?

Comment (colesbury) [reply torch/csrc/autograd/python_variable.cpp]: I'd suggest `T`. In general, we've tried to make the C++ Tensor API match the Python API even when it means uncommon names.

**Comment by ezyang** [comment]
This broke Windows:

```
20:36:33 ======================================================================
20:36:33 FAIL: test_inplace_view_saved_output (__main__.TestAutograd)
20:36:33 ----------------------------------------------------------------------
20:36:33 Traceback (most recent call last):
20:36:33   File "test_autograd.py", line 2817, in test_inplace_view_saved_output
20:36:33     self.assertEqual(dealloc[0], 1)
20:36:33   File "C:\Jenkins\workspace\pytorch-builds\pytorch-win-ws2016-cuda9-cudnn7-py3-test2\test\common_utils.py", line 528, in assertEqual
20:36:33     super(TestCase, self).assertLessEqual(abs(x - y), prec, message)
20:36:33 AssertionError: 1 not less than or equal to 1e-05 : 
20:36:33 
20:36:33 ----------------------------------------------------------------------
```

**Comment by kostmo** [comment]
This PR was [reverted](https://github.com/pytorch/pytorch/commit/0544a491d58f3c962a9ca88d31d182c6dd201a5c) due to breakage of `test_autograd`[on `pytorch-win-ws2016-cuda9-cudnn7-py3-test2`](https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-win-ws2016-cuda9-cudnn7-py3-test2/33541//console).

**Pull request closed without merging**